ambry-commons/src/test/java/com.github.ambry.commons/HelixNotifierTest.java

```diff
@@ -0,0 +1,365 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.commons;
+
+import com.github.ambry.config.HelixPropertyStoreConfig;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicReference;
+import org.apache.helix.ZNRecord;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import static com.github.ambry.commons.TestUtils.*;
+import static org.junit.Assert.*;
+
+
+/**
+ * Unit tests for {@link HelixNotifier}.
+ */
+public class HelixNotifierTest {
+  private static int zkClientConnectTimeoutMs = 20000;
+  private static int zkClientSessionTimeoutMs = 20000;
+  private static String zkClientConnectString = "localhost:2182";
+  private static String storeRootPath = "/ambry/testCluster/helixPropertyStore";
+
+  private static final HelixPropertyStoreFactory<ZNRecord> storeFactory = new MockHelixPropertyStoreFactory<>();
+  private static final List<String> refTopics = Arrays.asList(new String[]{"Topic1", "Topic2"});
+  private static final List<String> refMessages = Arrays.asList(new String[]{"Message1", "Message2"});
+  private final List<String> receivedTopicsByListener0 = new ArrayList<>();
+  private final List<String> receivedTopicsByListener1 = new ArrayList<>();
+  private final List<String> receivedMessagesByListener0 = new ArrayList<>();
+  private final List<String> receivedMessagesByListener1 = new ArrayList<>();
+  private final AtomicReference<CountDownLatch> latch0 = new AtomicReference<>();
+  private final AtomicReference<CountDownLatch> latch1 = new AtomicReference<>();
+  private static HelixPropertyStoreConfig storeConfig;
+  private static long latchTimeoutMs = 1000;
+  private HelixNotifier notifier;
+  private List<TopicListener<String>> listeners;
```

Comment:
this can be `final` - instead of using `setup()`, use a constructor - they are the same but using a constructor allows you to declare some variables as `final`